### PR TITLE
FIX-#191: Fix pendinq response after blocking GET request

### LINKED ISSUE #191: [MPI] Incorrect data response after receiving a blocking GET request

### PR DESCRIPTION
**Pull request by Retribution98**
Signed-off-by: Kirill Suvorov <kirill.suvorov@intel.com>

## What do these changes do?

- [x] passes `flake8 .`
- [x] passes `black .`
- [x] signed commit with `git commit -s` 
- [x] Resolves #191 
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: unidist/core/backends/mpi/core/worker/request_store.py [hunk 1]
```diff
@@ -50,7 +52,7 @@ def get_instance(cls):
             cls.__instance = RequestStore()
         return cls.__instance
 
-    def put(self, data_id, rank, request_type):
+    def put(self, data_id, rank, request_type, is_blocking_op=False):
```
Comment (YarShev) [inline unidist/core/backends/mpi/core/worker/request_store.py:55]: add a docstring for the parameter

File: unidist/core/backends/mpi/core/worker/request_store.py [hunk 2]
```diff
@@ -110,21 +115,27 @@ def check_pending_get_requests(self, data_ids):
         data_id : iterable or unidist.core.backends.common.data_id.DataID
             An ID or list of IDs to data.
         """
+
+        def check_data_id(data_id):
```
Comment (YarShev) [inline unidist/core/backends/mpi/core/worker/request_store.py:119]:
```suggestion
        def check_request(data_id):
```

File: unidist/core/backends/mpi/core/worker/request_store.py [hunk 3]
```diff
@@ -110,21 +115,27 @@ def check_pending_get_requests(self, data_ids):
...
+            if data_id in self._data_request:
+                ranks_with_get_request = self._data_request[data_id]
+                for rank_num in ranks_with_get_request:
+                    # Data is already in DataMap, so not problem here
+                    self.process_get_request(rank_num, data_id, is_blocking_op=False)
+                del self._data_request[data_id]
+            #
+            if data_id in self._blocking_data_request:
```
Comment (YarShev) [inline unidist/core/backends/mpi/core/worker/request_store.py:127]:
```suggestion
            elif data_id in self._blocking_data_request:
```

File: unidist/test/test_async_actor.py [hunk 4]
```diff
@@ -131,3 +132,27 @@ def foo(actor):
...
+
+
+@pytest.mark.skipif(
+    Backend.get() == BackendName.MP,
+    reason="Run of a remote task inside of another one is not implemented yet for multiprocessing",
+)
+@pytest.mark.skipif(
+    Backend.get() == BackendName.PY,
+    reason="Use courutine result is not implemented yet for python",
+)
```
Comment (YarShev) [inline unidist/test/test_async_actor.py:144]:
```suggestion
@pytest.mark.skipif(
    Backend.get() in (BackendName.MP, BackendName.PY),
    reason="MP and PY backends do not support execution of coroutines.",
)
```

File: unidist/core/backends/mpi/core/worker/request_store.py [hunk 5]
```diff
@@ -110,21 +115,27 @@ def check_pending_get_requests(self, data_ids):
...
+
         if isinstance(data_ids, (list, tuple)):
             for data_id in data_ids:
-                if data_id in self._data_request:
-                    ranks_with_get_request = self._data_request[data_id]
-                    for rank_num in ranks_with_get_request:
-                        # Data is already in DataMap, so not problem here
-                        self.process_get_request(rank_num, data_id)
-                    del self._data_request[data_id]
+                check_data_id(data_id)
```
Comment (YarShev) [inline unidist/core/backends/mpi/core/worker/request_store.py:136]:
```suggestion
                check_request(data_id)
```

File: unidist/core/backends/mpi/core/worker/request_store.py [hunk 6]
```diff
@@ -110,21 +115,27 @@ def check_pending_get_requests(self, data_ids):
...
         else:
-            if data_ids in self._data_request:
-                ranks_with_get_request = self._data_request[data_ids]
-                for rank_num in ranks_with_get_request:
-                    # Data is already in DataMap, so not problem here
-                    self.process_get_request(rank_num, data_ids)
-                del self._data_request[data_ids]
+            check_data_id(data_ids)
```
Comment (YarShev) [inline unidist/core/backends/mpi/core/worker/request_store.py:138]:
```suggestion
            check_request(data_ids)
```

File: unidist/core/backends/mpi/core/worker/request_store.py (hunk 3 above)
Comment (Retribution98) [reply unidist/core/backends/mpi/core/worker/request_store.py:127]: The data_id can be requested by both a blocking request and a non-blocking request, so we need to check in both lists.

**Review by YarShev — APPROVED** [review]

**Pull request merged**